nexuse2e-webapp/src/main/frontend/src/app/login/login.component.ts

```diff
@@ -0,0 +1,69 @@
+import { Component, Injectable, OnInit } from "@angular/core";
+import { ActivatedRoute, Router } from "@angular/router";
+import { DataService } from "../data/data.service";
+import { CacheService } from "../data/cache.service";
+import { NavigationService } from "../navigation/navigation.service";
+
+interface LoginData {
+  user: string;
+  password: string;
+}
+
+@Component({
+  selector: "app-login",
+  templateUrl: "./login.component.html",
+  styleUrls: ["./login.component.scss"],
+})
+@Injectable({ providedIn: "root" })
+export class LoginComponent implements OnInit {
+  returnUrl!: string;
+  isHttps?: boolean;
+  machineName?: string;
+  loginError?: boolean;
+  user?: string;
+  password?: string;
+
+  constructor(
+    private route: ActivatedRoute,
+    private router: Router,
+    private dataService: DataService,
+    private cacheService: CacheService,
+    private navigationService: NavigationService
+  ) {}
+
+  async ngOnInit() {
+    try {
+      await this.dataService.get("/logged-in");
+      await this.router.navigate(["/dashboard"]);
+    } catch {
+      const returnUrl = "returnUrl";
```

Comment:
Is having this const here giving any value? I think we should remove it or call it something like `returnUrlKeyName`.